src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingPlantation.java

```diff
@@ -86,317 +71,415 @@ public BuildingPlantation(final IColony c, final BlockPos l)
-     *
-     * @return a list of plants.
+     * TODO: future
+     * Legacy code, can be removed when plantations will no longer have to support fields
+     * directly from the hut building.
      */
-    public List<Item> getAvailablePlants()
+    @Override
+    public void setTileEntity(final AbstractTileEntityColonyBuilding te)
     {
-        final String setting = getSetting(MODE).getValue();
-
-        List<Item> items = new ArrayList<>();
-        if (setting.contains(Items.SUGAR_CANE.getDescriptionId()))
-        {
-            items.add(Items.SUGAR_CANE);
-        }
-        if (setting.contains(Items.CACTUS.getDescriptionId()))
-        {
-            items.add(Items.CACTUS);
-        }
-        if (setting.contains(Items.BAMBOO.getDescriptionId()))
+        super.setTileEntity(te);
+        if (triggerFieldMigration)
         {
-            items.add(Items.BAMBOO);
+            updateFields();
+            triggerFieldMigration = false;
         }
+    }
 
-        // Add sugar cane as the default plant cycle as a fallback in case nothing exists in the settings
-        if (items.isEmpty())
-        {
-            items.add(Items.SUGAR_CANE);
-            Log.getLogger().warn("Plantation plant setting contains none of the preconfigured plants, please report this to the developers!");
-        }
-        return items;
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return PLANTATION;
     }
 
-    public static class CraftingModule extends AbstractCraftingBuildingModule.Crafting
+    /**
+     * Field module implementation for the plantation.
+     */
+    public static class PlantationFieldsModule extends FieldsModule
     {
+        @Override
+        public void serializeToView(final @NotNull FriendlyByteBuf buf)
+        {
+            super.serializeToView(buf);
+            buf.writeInt(getMaxConcurrentPlants());
+        }
+
         /**
-         * Create a new module.
+         * Get the maximum allowed plants the plantation can work on simultaneously.
          *
-         * @param jobEntry the entry of the job.
+         * @return the maximum amount of concurrent plants.
          */
-        public CraftingModule(final JobEntry jobEntry)
+        public int getMaxConcurrentPlants()
         {
-            super(jobEntry);
+            return (int) Math.ceil(building.getBuildingLevel() / 2D);
         }
 
-        @NotNull
         @Override
-        public OptionalPredicate<ItemStack> getIngredientValidator()
+        protected int getMaxFieldCount()
         {
-            return CraftingUtils.getIngredientValidatorBasedOnTags(CRAFTING_PLANTATION).combine(super.getIngredientValidator());
+            int allowedPlants = (int) Math.ceil(building.getBuildingLevel() / 2D);
+            return building.getColony().getResearchManager().getResearchEffects().getEffectStrength(PLANTATION_LARGE) > 0
+                     ? allowedPlants + 1
+                     : allowedPlants;
         }
```

Comment:
Was this intended to be multiplication rather than division?  Because currently it's limiting to 1 plant and 2 fields regardless of level.

Also, if doing the research only adds a single extra field, it doesn't seem very worthwhile (especially if each level does add 20 fields, if it had been multiplication -- which also seems wrong).

Reply:
Yes, this works as intended. The intended field limits are: `1 1 2 2 3`. This is what the division is for, for example level 3: `3 / 2 = 1.5 = 2`.

The research adds an extra field across the board, giving you access to 4 fields at max, you're confused with planting positions per field, each field can have a max amount of work tags, which is what you got the 20 from.

Reply:
Ohhh, that's a `D`.  I read it as `20`.  Might have been better to write it as `2.0` instead.

Reply:
2.0 is the notation for a float no? Reason I used D is to have it cast to a double.

Reply:
No.  `2.0` is a double unless you add an `F` at the end.  It's pretty rare to use the `D` suffix.

Reply:
For some reason I had in my mind that writing decimal points in Java caused it to be a float, maybe am confused with C# here.

Reply:
All the C-based languages work the same as Java in that respect.

Reply:
I don't think float would matter in this case, as floats are cast to double automatically by java, when calling `Math.ceil`